frontend/src/components/shared/roomCardModal/RoomCardModalButton.tsx

```diff
@@ -1,21 +1,37 @@
+import { useState } from "react";
 import { useNavigate } from "react-router-dom";
 import useMutateRoom from "@/hooks/mutations/useMutateRoom";
 import Button from "@/components/common/button/Button";
-import { RoomInfo } from "@/@types/roomInfo";
+import { Role, RoomInfo } from "@/@types/roomInfo";
 
 interface RoomCardModalButtonProps {
   roomInfo: RoomInfo;
 }
 
 const RoomCardModalButton = ({ roomInfo }: RoomCardModalButtonProps) => {
   const navigate = useNavigate();
+  const [userRole, setUserRole] = useState<Role>("BOTH");
   const { postParticipateInMutation } = useMutateRoom();
   const isLoggedIn = !!localStorage.getItem("accessToken");
 
   const handleParticipateRoomClick = () => {
-    postParticipateInMutation.mutate(roomInfo.id, {
-      onSuccess: () => navigate(`/rooms/${roomInfo.id}`),
-    });
+    postParticipateInMutation.mutate(
+      {
+        roomId: roomInfo.id,
+        role: userRole as Role,
+      },
+      {
+        onSuccess: () => navigate(`/rooms/${roomInfo.id}`),
+      },
+    );
+  };
+
+  const handleRoleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      setUserRole("REVIEWER");
+    } else {
+      setUserRole("BOTH");
+    }
```

Comment:
```suggestion
    event.target.checked ? setUserRole("REVIEWER") : setUserRole("BOTH");
```
이렇게 해볼 수 있지 않을까요? 개인적으로 저는 이 방법을 선호해서 말씀드려봅니다 ㅎ

Reply:
넵 변경하였습니다!! 코드가 훨씬 간결해졌네요ㅎㅎ